ten-simple-rules-dockerfiles.Rmd

```diff
@@ -269,17 +272,18 @@ You can identify these images by the missing the user portion of the image name,
 However, these images only provide basic programming languages or very widely used software, so you will likely use images maintained by organizations or fellow researchers.
 While some organizations can be trusted to update containers with security fixes (see list below), for most individual accounts that provide ready to use images, it is likely that these will not be updated regularly.
 It is even possible that images or `Dockerfile`s may disappear, or that images are published with malicious intent, though we have not heard of any such case in academia.
-Therefore, for security, transparency, and reproducibility, you should only use images where you have access to the `Dockerfile`, and it is suggested to save a copy of the `Dockerfile` in the case that a repository or other distribution goes away.
+Therefore, for security, transparency, and reproducibility, you should only use images where you have access to the `Dockerfile`, e.g., by using automated builds on Docker Hub.
```

Comment:
I was not aware of the hack and change in practice. Will remove, but I guess mentioning the reason for it goes beyond the scope?

Reply:
There were some smaller ones, but this is the one https://www.zdnet.com/article/docker-hub-hack-exposed-data-of-190000-users/ that at least for me, made me step back and decide to use Quay.io instead, because you can generate repository specific robot tokens. Docker Hub has since added tokens for user login, but that's still a global token that (if discovered) would compromise a user's entire account. So these days I might push containers there directly, but most of the containers I care about are on Quay.io where I have better control of permissions.

I think that particular "Which registry is better" is out of scope for this paper, and it doesn't logically fit here. I think we should just say:

>Therefore, for security, transparency, and reproducibility, you should only use images where you have access to the `Dockerfile`.

It's implied that the images have to come from some registry.

Reply:
And it's not some official, declared change in practice, but (I tended to see) people shirk a bit.

Reply:
Thanks for the pointer! Will push an update later, and will have to rethink my whole routine...